Pod/Classes/SeafActivityModel.m

```diff
@@ -10,93 +10,62 @@
-            NSString *obj_type = [dict objectForKey:@"obj_type"];
-            if ([name containsString:@"(draft).md"]) {
-                obj_type = @"draft";
-            }
-            
-            NSString *opsKey = [NSString stringWithFormat:@"%@ %@", op_type, obj_type];
-            NSDictionary *opsMap = [NSDictionary dictionaryWithObjects:values forKeys:keys];
-            self.operation = [opsMap objectForKey:opsKey];
-            
-            if ([op_type isEqualToString:@"rename"]) {
-                if ([obj_type isEqualToString:@"file"]) {
-                    self.detail = [NSString stringWithFormat:@"%@ => %@", [dict objectForKey:@"old_name"], [dict objectForKey:@"name"]];
-                } else {
-                    NSString *old_key = [NSString stringWithFormat:@"old_%@_name", obj_type];
-                    NSString *key = [NSString stringWithFormat:@"%@_name", obj_type];
-                    self.detail = [NSString stringWithFormat:@"%@ => %@", [dict objectForKey:old_key], [dict objectForKey:key]];
-                }
-            } else if ([op_type isEqualToString:@"move"]) {
-                self.detail = [NSString stringWithFormat:@"%@ => %@", [dict objectForKey:@"old_path"], [dict objectForKey:@"path"]];
-            } else if ([op_type isEqualToString:@"clean-up-trash"]) {
-                NSInteger days = [[dict objectForKey:@"days"] integerValue];
-                if (days == 0) {
-                    self.operation = NSLocalizedString(@"Removed all items from trash", @"Seafile");
-                } else {
-                    self.operation = [NSString stringWithFormat:NSLocalizedString(@"Removed items older than %ld days from trash", @"Seafile"), days];
-                }
-                self.detail = [dict objectForKey:@"repo_name"];
+        self.operation = [self getOpreationFromOpType:opType objType:objType opsMap:opsMap even:even];
+        self.detail = [self eventRenamed:even opType:opType objType:objType];
+    }
+    return self;
+}
+
+- (NSString *)getOpreationFromOpType:(NSString *)opType objType:(NSString *)objType opsMap:(NSDictionary *)opsMap even:(NSDictionary *)even {
+    NSString *operation;
+    if (opType && objType) {
+        NSString *opsKey = [NSString stringWithFormat:@"%@ %@", opType, objType];
+        operation = [opsMap objectForKey:opsKey];
+        //clean-up-trash operation
+        if ([opType isEqualToString:@"clean-up-trash"]) {
+            NSString *days = [even objectForKey:@"days"];
+            if ([days integerValue] == 0) {
+                operation = NSLocalizedString(@"Removed all items from trash", @"Seafile");
+            } else {
+                operation = [NSString stringWithFormat:NSLocalizedString(@"Removed items older than %@ days from trash", @"Seafile"), days];
             }
         }
     }
-    return self;
+    return operation;
+}
+
+- (NSString *)eventRenamed:(NSDictionary *)even opType:(NSString *)opType objType:(NSString *)objType {
```

Comment:
why this function is called "eventRenamed"?
I think what it does is calculate the detail string?